cognite/client/data_classes/datapoints.py

```diff
@@ -116,6 +127,122 @@ class StatusCode(IntEnum):
+
+    @property
+    def status_code(self) -> NoReturn:
+        raise AttributeError(
+            "'MinDatapoint' object has no attribute 'status_code'. Tip: fetch using `include_status=True`."
+        )
+
+    @property
+    def status_symbol(self) -> NoReturn:
+        raise AttributeError(
+            "'MinDatapoint' object has no attribute 'status_symbol'. Tip: fetch using `include_status=True`."
+        )
+
+    @classmethod
+    def load(cls, dct: dict[str, Any]) -> Self:
+        assert "statusCode" not in dct
+        return cls(dct["timestamp"], dct["value"])
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {"timestamp": self.timestamp, "value": self.value}
+
+
+@dataclass(slots=True, frozen=True)
+class MinDatapointWithStatus(MaxOrMinDatapoint):
+    timestamp: int
+    value: float
+    status_code: int
+    status_symbol: str
+
+    @classmethod
+    def load(cls, dct: dict[str, Any]) -> Self:
+        return cls(dct["timestamp"], dct["value"], dct["statusCode"], dct["statusSymbol"])
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {
+            "timestamp": self.timestamp,
+            "value": self.value,
+            "statusCode" if camel_case else "status_code": self.status_code,
+            "statusSymbol" if camel_case else "status_symbol": self.status_symbol,
+        }
+
+
+@dataclass(slots=True, frozen=True)
+class MaxDatapoint(MaxOrMinDatapoint):
+    timestamp: int
+    value: float
+
+    @property
+    def status_code(self) -> NoReturn:
+        raise AttributeError(
+            "'MaxDatapoint' object has no attribute 'status_code'. Tip: fetch using `include_status=True`."
+        )
+
+    @property
+    def status_symbol(self) -> NoReturn:
+        raise AttributeError(
+            "'MaxDatapoint' object has no attribute 'status_symbol'. Tip: fetch using `include_status=True`."
+        )
+
+    @classmethod
+    def load(cls, dct: dict[str, Any]) -> Self:
+        assert "statusCode" not in dct
+        return cls(dct["timestamp"], dct["value"])
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {"timestamp": self.timestamp, "value": self.value}
+
+
+@dataclass(slots=True, frozen=True)
+class MaxDatapointWithStatus(MaxOrMinDatapoint):
+    timestamp: int
+    value: float
+    status_code: int
+    status_symbol: str
+
+    @classmethod
+    def load(cls, dct: dict[str, Any]) -> Self:
+        return cls(dct["timestamp"], dct["value"], dct["statusCode"], dct["statusSymbol"])
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {
+            "timestamp": self.timestamp,
+            "value": self.value,
+            "statusCode" if camel_case else "status_code": self.status_code,
+            "statusSymbol" if camel_case else "status_symbol": self.status_symbol,
+        }
+
+
+def select_min_or_max_datapoint_cls(dct: dict[str, Any], is_minimum: bool) -> type[MaxOrMinDatapoint]:
```

Comment:
Why not private? Only used in _load methods.

Reply:
Agree, will update